File: service/src/main/java/com/boxfox/cross/service/auth/AuthUsecase.java
package com.boxfox.cross.service.auth;

import com.linkbit.android.entity.UserModel;
import io.reactivex.Completable;
import io.reactivex.Single;

public interface AuthUsecase {
    Single<UserModel> signin(String token);
    Single<UserModel> getAccountByUid(String uid);
    Completable unRegister(String uid);
}

File: service/src/main/java/com/boxfox/cross/service/auth/AuthUsecase.kt
package com.boxfox.cross.service.auth

import com.boxfox.cross.common.entity.UserModel
import org.jooq.DSLContext

interface AuthUsecase {
    fun signin(ctx: DSLContext, token: String): UserModel
    fun getAccountByUid(ctx: DSLContext, uid: String): UserModel
    fun unRegister(ctx: DSLContext, uid: String)
}
